src/Frontend/Page/Idea.hs

```diff
@@ -270,13 +270,12 @@ instance FormPage EditIdea where
         <*> ("idea-category" .: makeFormSelectCategory (idea ^. ideaCategory))
         <*> pure (idea ^. ideaLocation)
 
-    -- FIXME: factor out code common with CreateIdea.
-    -- FIXME: category choice should look like in CreateIdea.
-    formPage = createOrEditPage True
+    formPage v form p@(EditIdea idea) = createOrEditPage True (idea ^. ideaLocation) v form p
 
 createOrEditPage :: (Monad m, Typeable page, Page page) =>
-    Bool -> View (HtmlT m ()) -> (HtmlT m () -> HtmlT m ()) -> page -> HtmlT m ()
-createOrEditPage showDeleteButton v form p = semanticDiv p $ do
+    Bool -> IdeaLocation ->
+    View (HtmlT m ()) -> (HtmlT m () -> HtmlT m ()) -> page -> HtmlT m ()
+createOrEditPage showDeleteButton cancelUrl v form p = semanticDiv p $ do
```

Comment:
Yes. On the other hand I am agree with Connor McBride, we should introduce different types for these parameters, even if at the beginning it is isomorphic to `Bool`. But I can live with the current solution.